src/v/raft/consensus.cc

```diff
@@ -2195,23 +2203,19 @@ consensus::delta_for_truncation(const snapshot_metadata& metadata) {
     return model::offset_delta{delta};
 }
 
-ss::future<> consensus::truncate_to_latest_snapshot() {
+ss::future<> consensus::truncate_to_latest_snapshot(
+  std::optional<model::offset_delta> force_truncate_delta) {
     // we have to prefix truncate config manage at exactly last offset included
     // in snapshot as this is the offset of configuration included in snapshot
     // metadata.
-    //
-    // We truncate the log before truncating offset translator to wait for
-    // readers that started reading from the start of the log before we advanced
-    // _last_snapshot_index and thus can still need offset translation info.
     return _log
       ->truncate_prefix(storage::truncate_prefix_config(
-        model::next_offset(_last_snapshot_index), _scheduling.default_iopc))
+        model::next_offset(_last_snapshot_index),
```

Comment:
this is invoked by hydrate_snapshot, but there the prefix truncate reset was being called with `_last_snapshot_index` rather than `next_offset(_last_snapshot_index)`. is that a material difference?

Reply:
Note that the log impl's prefix truncate will call offset_translator_reset() with `prev(truncate_at)`, which accounts for the difference, unless you noticed that and there's still a material difference